cli/azd/internal/figspec/fig_generators.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package figspec
+
+// Fig generator names used for dynamic autocomplete suggestions.
+// These constants map to the TypeScript generator implementations in the rendered Fig spec.
+const (
+	// FigGenListEnvironments generates suggestions from available azd environments
+	FigGenListEnvironments = "azdGenerators.listEnvironments"
+
+	// FigGenListEnvironmentVariables generates suggestions from environment variables
+	FigGenListEnvironmentVariables = "azdGenerators.listEnvironmentVariables"
+
+	// FigGenListTemplates generates suggestions from available azd templates
+	FigGenListTemplates = "azdGenerators.listTemplates"
+
+	// FigGenListTemplateTags generates suggestions from all available template tags
+	FigGenListTemplateTags = "azdGenerators.listTemplateTags"
+
+	// FigGenListTemplatesFiltered generates suggestions from templates filtered by --filter flag
+	FigGenListTemplatesFiltered = "azdGenerators.listTemplatesFiltered"
+
+	// FigGenListExtensions generates suggestions from all available extensions
+	FigGenListExtensions = "azdGenerators.listExtensions"
+
+	// FigGenListInstalledExtensions generates suggestions from installed extensions only
+	FigGenListInstalledExtensions = "azdGenerators.listInstalledExtensions"
+)
+
+// figGeneratorDefinitions returns the TypeScript code defining all Fig generators and interfaces used by them.
+// The generator names in this code must match the constants defined above.
+func figGeneratorDefinitions() string {
+	return `interface AzdEnvListItem {
```

Comment:
nit: Instead of inlining this TypeScript stuff could we save this as an embedded resource and return it from there? It would be easier to catch TS syntax issues/errors can get better IntelliSense during updates?